lib/Optimizer/Lower/Lower.cpp

```diff
@@ -448,6 +448,27 @@ static void lowerReluNode(Function *F, CompilationContext &cctx,
   replaceAllUsesOfWith(cctx.loweredInfoMap, R.getResult(), relu);
 }
 
+static void lowerHardSigmoidNode(Function *F, CompilationContext &cctx,
+                                 const HardSigmoidNode &R) {
+  LOG_SCOPE(F->getLogContext(), "lowerHardSigmoidNode")
+
+  auto ty = R.getResult().getType();
+  CHECK(ty->isFPType()) << "HardSigmoid: quantized type not supported: " << ty;
+
+  // max(0, min(1, alpha * x + beta))
+  auto *Alpha =
+      F->createSplat(DECORATE_NODE_NAME(R, "Alpha"), ty, R.getAlpha());
+  auto *Beta = F->createSplat(DECORATE_NODE_NAME(R, "Beta"), ty, R.getBeta());
+  auto *zero = F->createSplat(DECORATE_NODE_NAME(R, "zero"), ty, 0);
+  auto *one = F->createSplat(DECORATE_NODE_NAME(R, "one"), ty, 1);
+
+  auto *mul = F->createMul(DECORATE_NODE_NAME(R, "mul"), Alpha, R.getInput());
+  auto *add = F->createAdd(DECORATE_NODE_NAME(R, "add"), mul, Beta);
+  auto *min = F->createMin(DECORATE_NODE_NAME(R, "min"), add, one);
+  auto *max = F->createMax(DECORATE_NODE_NAME(R, "max"), min, zero);
```

Comment:
Please use `F->createClip()` here instead of creating individual min/max nodes. If the backend supports a Clip node then it's more efficient than creating the splats, and if it doesn't support Clip then it'll be lowered to splats like you have here.

Reply:
handled in `Graph.cpp`
https://github.com/pytorch/glow/blob/301b91d8d2935cf2ac1d48517d18cdac655c3d59/lib/Graph/Graph.cpp#L1301